File: typescript/test/action-response.test.ts
import { strict as assert } from "node:assert";
import { test } from "node:test";

import {
  MOBIUS_ACTION_CONTENT_TYPE,
  type ActionResponseEnvelope,
} from "../src/index.js";

test("action response envelope uses the canonical content type and shape", () => {
  const response: ActionResponseEnvelope<{ ok: boolean }> = {
    output: { ok: true },
    context: [{ name: "board", content: "fresh" }],
  };

  assert.equal(
    MOBIUS_ACTION_CONTENT_TYPE,
    "application/vnd.mobius.action+json",
  );
  assert.deepEqual(response, {
    output: { ok: true },
    context: [{ name: "board", content: "fresh" }],
  });
});

File: mobius/action_response_test.go
package mobius

import (
	"encoding/json"
	"testing"
)

func TestActionResponseEnvelope(t *testing.T) {
	if ActionResponseContentType != "application/vnd.mobius.action+json" {
		t.Fatalf("content type = %q", ActionResponseContentType)
	}
	body, err := json.Marshal(ActionResponseEnvelope{
		Output:  map[string]any{"ok": true},
		Context: []RuntimeContextItem{{Name: "board", Content: "fresh"}},
	})
	if err != nil {
		t.Fatal(err)
	}
	if got, want := string(body), `{"output":{"ok":true},"context":[{"content":"fresh","name":"board"}]}`; got != want {
		t.Fatalf("body = %s, want %s", got, want)
	}
}
